jabsrv/src/main/java/org/jabref/http/server/LibraryResource.java

```diff
@@ -94,16 +165,212 @@ public Response getBibtex(@PathParam("id") String id) {
+        }
+
+        // TODO: Currently, the preview preferences are in GUI package, which is not accessible here.
+        // build the preview
+        BibEntry entry = entriesByCitationKey.getFirst();
+
+        String author = entry.getField(StandardField.AUTHOR).orElse("(N/A)");
+        String title = entry.getField(StandardField.TITLE).orElse("(N/A)");
+        String journal = entry.getField(StandardField.JOURNAL).orElse("(N/A)");
+        String volume = entry.getField(StandardField.VOLUME).orElse("(N/A)");
+        String number = entry.getField(StandardField.NUMBER).orElse("(N/A)");
+        String pages = entry.getField(StandardField.PAGES).orElse("(N/A)");
+        String releaseDate = entry.getField(StandardField.DATE).orElse("(N/A)");
+
+        // the only difference to the plain text version of this method is the format of the output:
+        String preview =
+                "<strong>Author:</strong> " + author + "<br>" +
+                        "<strong>Title:</strong> " + title + "<br>" +
+                        "<strong>Journal:</strong> " + journal + "<br>" +
+                        "<strong>Volume:</strong> " + volume + "<br>" +
+                        "<strong>Number:</strong> " + number + "<br>" +
+                        "<strong>Pages:</strong> " + pages + "<br>" +
+                        "<strong>Released on:</strong> " + releaseDate;
+
+        return preview;
+    }
+
+    /**
+     * At libraries/{id}/entries/pdffiles <br><br>
+     *
+     * Loops through all entries in the specified library and adds attached files of type "PDF" to
+     * a list and JSON serialises it.
+     */
+    @GET
+    @Path("entries/pdffiles")
+    @Produces(MediaType.APPLICATION_JSON + ";charset=UTF-8")
+    public String getPDFFilesAsList(@PathParam("id") String id) throws IOException {
+        // get a list of all entries in library (specified by "id")
+        BibDatabaseContext databaseContext = getDatabaseContext(id);
+        List<LinkedPdfFileDTO> response = new ArrayList<>();
+        List<BibEntry> entries = databaseContext.getDatabase().getEntries();
+        if (entries.isEmpty()) {
+            throw new NotFoundException("No entries found for library: " + id);
+        }
+
+        // loop through all entries to extract pdfs and paths
+        for (BibEntry entry : entries) {
+            List<LinkedFile> pathsToFiles = entry.getFiles();
+            if (!pathsToFiles.isEmpty()) {
+                for (LinkedFile file : pathsToFiles) {
+                    // ignore all non pdf files and online references
+                    if (!file.getFileType().equals("PDF") || LinkedFile.isOnlineLink(file.getLink())) {
+                        continue;
+                    }
+                    // add file to response body
+                    LinkedPdfFileDTO localPdfFile = new LinkedPdfFileDTO(entry, file);
+                    response.add(localPdfFile);
+                }
+            }
+        }
```

Comment:
A good excercise would be to convert this to a Java Stream - but this might be too much for now.